rfc/0023-kubewarden-stack-telemetry.md

```diff
@@ -0,0 +1,373 @@
+
+When the telemetry endpoint is configured, a new periodic job should be created
+in the controller to collect and send the data. This job can be implemented
+using the Runnable interface from the controller-runtime package and integrated
+into the manager alongside the existing reconcilers.
+
+This new periodic reconciler will start a time.Ticker that will periodically:
+
+- Collect information about how many PolicyServers are deployed
+- Collect information about how many policies are deployed
+- Collect the Kubewarden version in use (this can be a constant updated on
+  every release).
+- Collect the Kubernetes version.
+- Build a JSON payload with the collected data and send it to the remote server
+  using secure communication.
+- The controller logs the available updates for the Kubewarden version running returned by
+  the `upgrade-responder` server
+
+The execution of this new reconciler should never interrupt the functionality
+of the remaining reconcilers. This means that if the controller is
+misconfigured with invalid endpoint, the reconciler should log an error and
+continue.
+
+## Telemetry Server
+
+The telemetry server will be an instance of the [Longhorn
+upgrade-responder](https://github.com/longhorn/upgrade-responder).
+This server will receive information from controllers and store it in an
+InfluxDB database for further analysis.
+
+The server exposes an endpoint to receive a JSON payload with the following
+structure:
+
+```json
+{
+  "appVersion": "v1.28.0",
+  "extraTagInfo": {
+    "kubernetesVersion": "v1.33.0",
+    "instanceUid": "<unique-id-for-controller>"
+  },
+  "extraFieldInfo": {
+    "policyServerCount": 2,
+    "policiesCount": 3
+  }
+}
+```
+
+When the `upgrade-responder` receives a request, it will:
+
+1. Validate the payload using a predefined JSON schema.
+2. [Infer](https://github.com/longhorn/upgrade-responder/blob/a6f6c7736b7e420b07ae7d813765dac778ebc638/upgraderesponder/service.go#L509)
+   the origin's geo-location from the request's source IP address.
+3. Store the enriched data in an InfluxDB database.
+4. Respond the request with the available updates to the given Kubewarden version
+   defined in the `appVersion` field.
+
+The `upgrade-responder` will write the request data into an InfluxDB metric. The
+`extraTagInfo` values will be added as
+[tags](https://github.com/longhorn/upgrade-responder/blob/a6f6c7736b7e420b07ae7d813765dac778ebc638/upgraderesponder/service.go#L538),
+while the `extraFieldInfo` values will be added as
+[fields](https://github.com/longhorn/upgrade-responder/blob/a6f6c7736b7e420b07ae7d813765dac778ebc638/upgraderesponder/service.go#L558).
+
+For example, the JSON payload above would result in a data point with the
+following tags:
+
+- `appVersion`: The Kubewarden version (e.g., v1.28.0).
+- `kubernetesVersion`: The Kubernetes version where the controller is running
+  (e.g., v1.33.0).
+- `instanceUid`: A UID to uniquely identify the controller instance.
```

Comment:
I believe we need to explain how this `instanceUid` is unique. Is it unique per Deployment, or Kubewarden stack release?
I would expect it to be the second one, and consistent throughout upgrades. Hence I believe it should be a random UID value created on `helm install`, and saved as an opaque value in the Helm release ConfigMaps or Secrets

We should also mention it in its relevant section (for example "Helm chart changes").